### PR TITLE
Added missing check for port (inlet or outlet) bound to many topics.

### LINKED ISSUE #549: Fix: ensure you cannot connect an inlet or an outlet to many topics.

### PR DESCRIPTION
**Pull request by RayRoestenburg**
### What changes were proposed in this pull request?

Fixes #549  Missing check for connecting an inlet or outlet to more than one topic.

### Why are the changes needed?

The API implementation does not auto merge sources to the inlet, or broadcast to sinks from an outlet. So it is important this check is in place.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

Unit tests.

### CODE REVIEW

File: core/cloudflow-blueprint/src/main/scala/cloudflow/blueprint/Blueprint.scala [hunk 1]
```diff
@@ -215,6 +215,21 @@ final case class Blueprint(
     if (inlets.nonEmpty) problems = problems :+ UnconnectedInlets(inlets)
     problems
   }
+  private def verifyPortsBoundToManyTopics(
+      verifiedStreamlets: Vector[VerifiedStreamlet],
```
Comment (maasg) [inline core/cloudflow-blueprint/src/main/scala/cloudflow/blueprint/Blueprint.scala:219]: this param doesn't seem to be used in the impl/

File: core/cloudflow-blueprint/src/main/scala/cloudflow/blueprint/Blueprint.scala [hunk 2]
```diff
@@ -215,6 +215,21 @@ final case class Blueprint(
...
+      verifiedTopics: Vector[VerifiedTopic]
+  ): Vector[PortBoundToManyTopics] =
+    verifiedTopics
+      .flatMap(verifiedTopic => verifiedTopic.connections.map(_ -> verifiedTopic.id))
+      .groupBy { case (verifiedPort, _) => verifiedPort }
+      .flatMap {
+        case (verifiedPort, groupedTopicIds) =>
+          val topicIds = groupedTopicIds.map { case (_, topic) => topic }.toIndexedSeq
```
Comment (maasg) [inline core/cloudflow-blueprint/src/main/scala/cloudflow/blueprint/Blueprint.scala:227]:
```suggestion
          val topicIds = groupedTopicIds.map { case (_, topic) => topic }
```

File: core/cloudflow-blueprint/src/test/scala/cloudflow/blueprint/BlueprintSpec.scala [hunk 3]
```diff
@@ -252,7 +252,10 @@ class BlueprintSpec extends WordSpec with MustMatchers with EitherValues with Op
       blueprint.problems.collect { case unconnected: UnconnectedInlets ⇒ unconnected }.size mustBe 0
       val paths = Vector(VerifiedPortPath(ingressRef.name, "out"), VerifiedPortPath(egress2Ref.name, "in"))
         .sortBy(_.toString)
-      blueprint.problems mustBe Vector(
+      blueprint.problems.filterNot {
```
Comment (maasg) [inline core/cloudflow-blueprint/src/test/scala/cloudflow/blueprint/BlueprintSpec.scala:255]: this `filterNot`...`true` is so confusing.

Comment (RayRoestenburg) [reply core/cloudflow-blueprint/src/test/scala/cloudflow/blueprint/BlueprintSpec.scala]: flipped it for you.

File: core/cloudflow-blueprint/src/main/scala/cloudflow/blueprint/Blueprint.scala (hunk 1 above)
Comment (RayRoestenburg) [reply core/cloudflow-blueprint/src/main/scala/cloudflow/blueprint/Blueprint.scala]: Fixed.

Comment (RayRoestenburg) [reply core/cloudflow-blueprint/src/main/scala/cloudflow/blueprint/Blueprint.scala]: Good catch!

**Pull request merged**